### ISSUE TITLE
More info about MAXDOCTABLESIZE

### ISSUE THREAD

**Issue by botzill** [RediSearch/RediSearch#1527]
Hi.

Can you, please, let us know more info about this config, and when are use cases that we need to increase it? We will have hundreds of indexes and each index can have millions of docs. Do we need to increase this config? 

Thx.

**Comment by gkorland** [comment]
@botzill thanks for the question, we'll extend the MAXDOCTABLESIZE docs.

But, basically this configuration controls the internal Doc Table max size which is an "hashtable" (we'll probably deprecate it in the future and convert it to "standard" Hashtable. The Doc Table as it's implemented today is trying to avoid resharding by ever growing up to the MAXDOCTABLESIZE and only then stars to build lists in each entry in the table.

Bottom line, increasing this configuration can improve performance in case you expect more than 1M docs with some memory overhead, while decreasing it can save some memory in case you hold a small index that you keep update.

**Comment by botzill** [comment]
Hi @gkorland, thx for the update. 

The idea is that we want to have multiple indexes and each index can have from a few thousands of docs to millions. As I understand if we have an index with 100 docs and we have MAXDOCTABLESIZE 1M, it's really optimal? 

What would be the solution, in this case, to have a single index where we store all the data and have some flag to delimit individual docs for different use cases. The reason we want a separate index is that we may want to reset all the data, and deleting the index and creating again is faster than deleting each doc individually(when we have millions of docs).
